versioned_docs/version-4.6/foundations/harper-architecture.md

```diff
@@ -0,0 +1,93 @@
+---
+title: Harper Architecture
+---
+
+# Harper Architecture
+
+Before diving deep into APIs and configuration, it helps to understand the big picture of how Harper works.  
+Harper uses a **three-layer architecture** designed for distributed, edge-first computing. Each layer builds on the next, letting you start simple and scale as your app grows.
+
+![](/img/v4.6/harper-architecture.png)
+
+At a high level:  
+- **Core services** handle data, networking, and files.  
+- **Extensions** layer in reusable features (REST, GraphQL, Next.js, etc.).  
+- **Applications** bring everything together to deliver user-facing functionality.  
+
+:::info
+💡 **Why it matters:** You focus on building your app, while Harper takes care of scaling, networking, and consistency behind the scenes.
+:::
+
+---
+
+## Core Services
+
+Harper ships with three essential services:
+
+- **Database** → Fast storage, queries, and transactions.  
+- **Networking** → REST/HTTP, WebSockets, MQTT, and cluster communication.  
+- **File system** → File operations and serving static assets.  
+
+Think of these as Harper’s foundation—every extension and app builds on them.
+
+---
+
+## Applications & Extensions
+
+Most of your work will happen here.  
+
+### Applications  
+Applications sit at the top layer. They’re where you implement user-facing features. Examples:  
+- A **Next.js app** served directly from Harper.  
+- An **Apollo GraphQL server** exposing APIs.  
+
```

Comment:
Lets drop the Apollo example and instead detail something like the getting started guide builds. Basic app that exposes REST endpoints